src/Enzo/enzo_EnzoRiemannImpl.hpp

```diff
@@ -428,92 +375,55 @@ void EnzoRiemannImpl<ImplFunctor>::solve
 
   if (LUT::has_bfields()){
     // If Dedner Fluxes are required, they might get handled here
     //   - It would probably be better to handle this separately from the
     //     Riemann Solver since we already precompute L & R conserved
     //     AND it doesn't require wavespeed information.
-  } else {
-    // TODO: finish refactoring the VL+CT method so it can be ran entirely
-    //        without magnetic fields (so that we can remove the following):
-    if (force_bfield_fluxes_to_zero_){
-      set_bfield_fluxes_to_zero_(array_factory, flux_group);
-    }
-  }
-}
-
-//----------------------------------------------------------------------
-
-inline void compute_unity_sum_passive_fluxes_(const enzo_float dens_flux,
-					      EFlt3DArray *flux_arrays,
-					      EFlt3DArray *reconstructed,
-					      const int num_fields,
-					      const int iz, const int iy,
-					      const int ix) noexcept
-{
-  ERROR("EnzoRiemannImpl::solve_passive_advection_",
-        "This codepath is untested");
-  enzo_float sum = 0.;
-  for (int field_ind=0; field_ind<num_fields; field_ind++){
-    sum += reconstructed[field_ind](iz,iy,ix);
-  }
-  for (int field_ind=0; field_ind<num_fields; field_ind++){
-    flux_arrays[field_ind](iz,iy,ix) = (reconstructed[field_ind](iz,iy,ix)
-					* dens_flux / sum);
   }
 }
 
 //----------------------------------------------------------------------
 
 inline void passive_advection_helper_
-(std::string group_name, Grouping &priml_group, Grouping &primr_group,
- Grouping &flux_group, const EFlt3DArray &density_flux, int dim,
- EnzoFieldArrayFactory array_factory, const bool unity_sum) noexcept
+(const str_vec_t &passive_keys,
+ EnzoEFltArrayMap& prim_map_l, EnzoEFltArrayMap& prim_map_r,
+ EnzoEFltArrayMap& flux_map, const EFlt3DArray &density_flux,
+ int stale_depth) noexcept
 {
-  int num_fields = priml_group.size(group_name);
-  if (num_fields == 0) {return;}
-  EnzoPermutedCoordinates coord(dim);
+  std::size_t num_keys = passive_keys.size();
+  if (num_keys == 0) {return;}
 
-  // This was basically transcribed from hydro_rk in Enzo:
+  // This was essentially transcribed from hydro_rk in Enzo:
 
   // load array of fields
-  EFlt3DArray *wl_arrays = new EFlt3DArray[num_fields];
-  EFlt3DArray *wr_arrays = new EFlt3DArray[num_fields];
-  EFlt3DArray *flux_arrays = new EFlt3DArray[num_fields];
-  
-  for (int field_ind=0; field_ind<num_fields; field_ind++){
-    wl_arrays[field_ind] = array_factory.reconstructed_field(priml_group,
-							     group_name,
-							     field_ind, dim);
-    wr_arrays[field_ind] = array_factory.reconstructed_field(primr_group,
-							     group_name,
-							     field_ind, dim);
-    flux_arrays[field_ind] = array_factory.from_grouping(flux_group,
-							 group_name,
-							 field_ind);
+  EFlt3DArray *wl_arrays = new EFlt3DArray[num_keys];
+  EFlt3DArray *wr_arrays = new EFlt3DArray[num_keys];
+  EFlt3DArray *flux_arrays = new EFlt3DArray[num_keys];
```

Comment:
```suggestion
  const EFlt3DArray *wl_arrays = new EFlt3DArray[num_keys];
  const EFlt3DArray *wr_arrays = new EFlt3DArray[num_keys];
  const EFlt3DArray *flux_arrays = new EFlt3DArray[num_keys];
```
This might work - in which case would be more robust.

Reply:
Yes, it should definitely work, since `CelloArray` has pointer semantics. 

The shape and the pointer to array memory of a `const CelloArray<T, D>` can't be changed, but the contents of the array still can be changed.

As an aside, I was inspired by Kokkos and I'm currently working on a PR to support `CelloArray<const T, D>`, which is equivalent to a pointer to a constant (I'm adding support for implicit casting from `CelloArray<T, D>` to `CelloArray<const T, D>` and making sure all of the methods behave properly).